### ISSUE TITLE
Last bit of file being marked read-only breaks refiling

### ISSUE THREAD

**Issue by mediapathic** [NicolasPetton/nroam#27]
I have a file that looks like 
```
* header1
* * sub1 
* * sub2

* header2 
* * sub3
* Backlinks
```
The "Backlinks" is, of course, being generated by nroam. If I try to refile (using visual refile) to `header2`, I'm told that "Text is read-only". I can refile fine to header1. I am 90% sure this is being caused by nroam. Somewhat oddly, I also get this error when trying to do an org-capture to a datetree file (so, the place that's being filed to is not at the end of the file), which is what gives me that 10% of doubt, but disabling nroam lets me refile again, so I'm pretty sure.

**Comment by NicolasPetton** [comment]
Yes, that sounds like something caused by nroam. I'll investigate.

**Comment by snuffop** [comment]
Instead of creating a new issue  I believe I have another symptom that (may?) be related..  when I try to use org-roam-capture for dailies.

```
(setq  org-roam-dailies-capture-templates
         '(
           ("d" "default" plain (function org-roam--capture-get-point)
            "%?"
            :file-name "daily/%<%Y-%m-%d>"
            ;; :head "#+TITLE: Daily Notes for %<%A, %B %d %Y>"
            :unnarrowed t)
           ("t" "Tasks" entry
            #'org-roam-capture--get-point
            "* TODO  %?"
            :file-name "daily/%<%Y-%m-%d>"
            :olp ("Tasks"))
           ("r" "Rackspace" entry
            #'org-roam-capture--get-point
            "* %<%H:%M>  %?"
            :file-name "daily/%<%Y-%m-%d>"
            :olp ("Rackspace"))
           ("j" "Journal" entry
            #'org-roam-capture--get-point
            "* %<%H:%M>  %?"
            :file-name "daily/%<%Y-%m-%d>"
            :olp ("Journal"))))
 ```

When I capture into my daily olp  I get Capture template 'j': Text is read-only  yes when I capture to my Rackspace it works perfectly.
now if I move the headings around where Rackspace is at the end of the file and tasks above it.. the reverse symptom happens where Capture Template 'r': Text is read-only.

and of course if I disable nroam-mode  everything works as it had before.

if this is not related I will create another issue.. but ...

**Comment by NicolasPetton** [comment]
@snuffop I believe it is indeed related. I've been thinking about how to get a proper fix for this this, but the issue it a bit tricky.

**Comment by snuffop** [comment]
Alright.. good in the fact that It's not additional noise and is further context around the issue.  Thank You.

**Comment by smeagol74** [comment]
I'm experiencing the same thing with journal type capture templates. 

What if let user to specify in which files nroam should be added and where not? 

For example, we may use the file-level tag `nonroam` like this `#+filetags: nonroam`. When nroam sees such file-level tag in the file just not add the nroam part to it.

This could help with such journaling captures, but not with the refiling issue of course.

**Comment by smeagol74** [comment]
Or even let user to configure which tags should block nroam file improvements.
